### ISSUE TITLE
mypy should give error for the type with `''` or `""` in a type hint

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20286]
*Memo:
- `mypy --strict test.py`
- mypy 1.18.2
- Python 3.14.0
- Windows 11

The type with `''` or `""` in a type hint doesn't get error as shown below and `''` and `""` aren't shown in [Type parameter lists](https://docs.python.org/3/reference/compound_stmts.html#type-parameter-lists) section:

```python
#  ↓   ↓
v: 'int' = 3 # No error
```

```python
#  ↓   ↓
v: "int" = 3 # No error
```

So, mypy should give error for it.

**Comment by A5rocks** [comment]
These are forward references and therefore allowed. Also, "type parameter lists" are not that.

Potentially mypy should warn about these not being necessary, but this issue isn't that.
